### DISCUSSION TITLE
xcode-project use-profiles failing due to certificate length

### DISCUSSION THREAD

**Discussion by pstromberg98** [codemagic-ci-cd/codemagic-docs#2663] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Run `xcode-project use-profiles` with a signing certificate that has a subject name greater than 64 characters.

### Expected results

For it to utilize the valid certificate.

### Actual results

Our signing action (which just uses xcode-project use-profiles) fails with:

```
ERROR: /Users/builder/.pyenv/versions/3.8.13/lib/python3.8/site-packages/codemagic/models/certificate.py:88: UserWarning: Attribute's length must be >= 1 and <= 64, but it was 66
  for name_attribute in self.certificate.subject
```

### Build id (optional)

66abcddaf98e86168c45e3b8

**Comment by helinanever** [comment]
Hi @pstromberg98 , thanks for reporting this. We are looking into it.

**Comment by priitlatt** [comment, marked as answer]
@pstromberg98, actually there is nothing wrong with the certificate. It is just that [RFC-5280](https://datatracker.ietf.org/doc/html/rfc5280.html) specifies min and max lengths for certificate's common name, which isn't respected/enforced by Apple. Recent version of [cryptography](https://cryptography.io/en/latest/) reports about such cases as a warning.

The actual failure reason is in your project file `ios/Runner.xcodeproj/project.pbxproj`, where code signing entitlements are not defined properly. According to our logs the problematic line is 463:
```
CODE_SIGN_ENTITLEMENTS = CODE_SIGN_ENTITLEMENTS = "Runner/RunnerDebug-production.entitlements";;
```
which I believe should instead be
```
CODE_SIGN_ENTITLEMENTS = "Runner/RunnerDebug-production.entitlements";
```

You can use the switch [--code-signing-setup-verbose-logging](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/xcode-project/use-profiles.md#--code-signing-setup-verbose-logging) with `xcode-project use-profiles` to see more detailed logs, including errors, about actual project updates that are carried out to set provisioning profiles.

**Comment by pstromberg98** [comment]
@priitlatt Ahh got it. I just tested reverting back to `use-profiles` in main where I already fixed the malformed pbx. Didn't realize that was the root cause of this problem as well 🤦🏻‍♂️. I might just turn on that verbose logging as the error output there made it seem like the name was the issue. Thanks!
